### PR TITLE
Added possibility to use another ValidationParser through the config

### PR DESCRIPTION
**Pull request by develth**
I wanted to use another ValidationParser instead of the default provided. 

So i added the possibility to configure it.

### CODE REVIEW

File: DependencyInjection/NelmioApiDocExtension.php [hunk 1]
```diff
@@ -56,10 +56,7 @@ public function load(array $configs, ContainerBuilder $container)
             $container->setParameter('nelmio_api_doc.sandbox.authentication', $config['sandbox']['authentication']);
         }
 
-        // backwards compatibility for Symfony2.1 https://github.com/nelmio/NelmioApiDocBundle/issues/231
-        if (!interface_exists('\Symfony\Component\Validator\MetadataFactoryInterface')) {
-            $container->setParameter('nelmio_api_doc.parser.validation_parser.class', 'Nelmio\ApiDocBundle\Parser\ValidationParserLegacy');
```
Comment (BigZ) [inline DependencyInjection/NelmioApiDocExtension.php]: you should document this as well (the fact that nelmioapidocbundle has 2 validatores you can choose from by default). people running symfony 2.1 might have troubles after upgrading

Comment (develth) [reply DependencyInjection/NelmioApiDocExtension.php]: I did that in:

https://github.com/nelmio/NelmioApiDocBundle/pull/1058/commits/0ea070348cd99cda455934d36bd1cd7ae649cf0d#diff-ec2c2f174cd03c16b4e39e899eb4e340

Or do you mean in Readme.md ?

File: Resources/doc/configuration-in-depth.rst [hunk 2]
```diff
@@ -94,6 +94,16 @@ registration:
...
 
+
+If you want to user another Validation Parser, or want backwards Compatibility for Symfony 2.1, change the validation parser config
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    nelmio_api_doc:
+        parser:
+            validation: MyBundl\Parser\MyValidationParser
```
Comment (BigZ) [inline Resources/doc/configuration-in-depth.rst]: i think ’Nelmio\ApiDocBundle\Parser\ValidationParserLegacy’ would be more insightful as an example !

**Comment by develth** [comment]
Will change it. Thanks.

**Comment by GuilhemN** [comment]
Closing as 2.0 is no longer maintained, for new features please target 3.0.

**Pull request closed without merging**